play-json/shared/src/main/scala/JsMacroImpl.scala

```diff
@@ -25,16 +25,16 @@ import scala.reflect.macros.blackbox
     }
   }
 
-  def formatImpl[A: c.WeakTypeTag]: c.Expr[OFormat[A]] =
-    macroImpl[A, OFormat, Format](
+  def formatImpl[A: c.WeakTypeTag, O <: Json.MacroOptions]: c.Expr[OFormat[A]] =
```

Comment:
You can make the implicit available as a property of the same type that defines the macro (ie, make it a val on `WithOptions`), and then you can access it via `c.prefix`. See #85 :)